runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java

```diff
@@ -296,4 +309,53 @@ private void drawTooltip(Graphics2D graphics, XpGlobe mouseOverSkill, int goalXp
 
 		xpTooltip.render(graphics);
 	}
+
+	private void drawLevels(Graphics2D graphics, int startXp, int goalXp, int x, int y)
+	{
+		final Collection<Double> levelPercentages = levelPercentagesCache.computeIfAbsent(
```

Comment:
This would make more sense if it was cached within the xpglobe. Also, this has potentially unbound growth.